src/ParallelAlgorithms/LinearSolver/AsynchronousSolvers/ElementActions.hpp

```diff
@@ -0,0 +1,236 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <tuple>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/PrefixHelpers.hpp"
+#include "IO/Observer/Helpers.hpp"
+#include "IO/Observer/ObservationId.hpp"
+#include "IO/Observer/ObserverComponent.hpp"
+#include "IO/Observer/ReductionActions.hpp"
+#include "Informer/Verbosity.hpp"
+#include "NumericalAlgorithms/Convergence/HasConverged.hpp"
+#include "NumericalAlgorithms/LinearSolver/Gmres.hpp"
+#include "NumericalAlgorithms/LinearSolver/InnerProduct.hpp"
+#include "Parallel/ConstGlobalCache.hpp"
+#include "Parallel/Invoke.hpp"
+#include "Parallel/Printf.hpp"
+#include "Parallel/Reduction.hpp"
+#include "ParallelAlgorithms/Initialization/MergeIntoDataBox.hpp"
+#include "ParallelAlgorithms/LinearSolver/Tags.hpp"
+#include "Utilities/GetOutput.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/Requires.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace tuples {
+template <typename...>
+class TaggedTuple;
+}  // namespace tuples
+/// \endcond
+
+namespace LinearSolver {
+namespace async_solvers_detail {
```

Comment:
I'm not sure I agree with the practice of putting actions and tags that appear in a databox in `detail` namespaces, even if they are useful only during preconditioning -- I think `detail` should mostly be reserved for more self-contained implementation code.

Reply:
Agreed, thanks for bringing this up. I pushed a fixup that makes this namespace and the Richardson tags public. Only the algorithm-specific actions that cannot stand on their own should remain in a detail namespace I think, since they are implementation details of the algorithm and exposed in a specific order in the linear solver class (e.g. as `Richardson::prepare_step`, `Richardson::perform_step`).
I carry over this change to the CG and GMRES linear solvers in another commit (if you prefer I can factor that one out in a separate PR as well but the diff is quite trivial).